pallets/loans/src/lib.rs

```diff
@@ -523,20 +543,20 @@ pub mod pallet {
 			ensure_signed(origin)?;
 
 			// try to write off
-			let index = Self::write_off_loan(pool_id, loan_id, None)?;
-			Self::deposit_event(Event::<T>::WrittenOff(pool_id, loan_id, index));
+			let (active_count, (index, percentage, penalty_interest_rate_per_sec)) =
+				Self::write_off_loan(pool_id, loan_id, WriteOffAction::WriteOffToCurrentGroup)?;
+			Self::deposit_event(Event::<T>::WrittenOff(
+				pool_id,
+				loan_id,
+				percentage,
+				penalty_interest_rate_per_sec,
+				index,
+			));
 
 			// since the write off group index is picked in loop sequentially,
 			// total loops = index+1
-
-			// this is soft limit, so runtime do not have enough in the deposit.
-			// charge the maximum we have deposited
-			let count = index + 1;
-			if count >= T::MaxWriteOffGroups::get() {
-				return Ok(().into());
-			}
-
-			Ok(Some(Self::write_off_group_weight(count as u64)).into())
+			let count = index.expect("Write offs should always return a valid index") + 1;
```

Comment:
Do we really want to `.expect()` here? 🤔

Reply:
and what if it overflows? I don't want a stale chain 😢

Reply:
It's an expect for sure here. The internal `write_off_loan` is shared by both `admin_write_off` and the regularly `write_off` extrinsic (since most of the logic is identical). The non-admin case always returns an index, and the admin case always does not.

It also will not overflow since the index is capped by `T::MaxWriteOffGroups`

I'll add a better comment here explaining all that